buildSrc/src/main/groovy/org/opencms/GwtCompiler.groovy

```diff
@@ -0,0 +1,203 @@
+        gwtExtraDir = project.buildDir.toString() + "/extra/$opencmsModuleName"
+        return this
+    }
+
+    @OutputDirectories
+    Set<File> getOutputDirs() {
+        [new File(gwtWarDir), new File(gwtExtraDir)]
+    }
+
+    @InputFiles
+    Set<File> getGwtFiles() {
+        sourceSetGwt.java.srcDirs
+    }
+
+    @InputDirectory
+    File getGwtResourceDir() {
+        sourceSetGwt.output.resourcesDir
+    }
+
+    /**
+     * Sets the 'OpenCms Core' source set.
+     *
+     * @param sourceSetMain the main source set
+     * return this
+     */
+    GwtCompiler sourceSetMain(SourceSet sourceSetMain) {
+        this.sourceSetMain = sourceSetMain
+        return this
+    }
+
+    /**
+     * Sets the 'OpenCms Modules' source set.
+     *
+     * @param sourceSetModules the modules source set
+     * return this
+     */
+    GwtCompiler sourceSetModules(SourceSet sourceSetModules) {
+        this.sourceSetModules = sourceSetModules
+        return this
+    }
+
+    /**
+     * Sets the OpenCms GWT source set.
+     *
+     * @param sourceSetGwt the gwt source set
+     * return this
+     */
+    GwtCompiler sourceSetGwt(SourceSet sourceSetGwt) {
+        this.sourceSetGwt = sourceSetGwt
+        return this
+    }
+
+    @Override
+    @TaskAction
+    void exec() {
+        println '====================================================================================================='
+        println "Building GWT resources for opencms module '$opencmsModuleName' (GWT module '$gwtModuleName')"
+        println '====================================================================================================='
+
+        // Do not clean output if in draft mode, to allow for incremental compilation
+        if (draft) {
+            logger.warn "GWT-Compiling DRAFT version. Do not use in production!"
+        } else {
+            // Clean the output directories
+            [gwtWarDir, gwtExtraDir].each { dirName ->
+                def dir = new File(dirName)
+                if (dir.exists()) {
+                    dir.delete()
+                    logger.info "dir $dir deleted"
+                }
+                dir.mkdirs()
+            }
+        }
+        logger.debug "war dir: $gwtWarDir"
+        logger.debug "extra dir: $gwtExtraDir"
+
+        if (logger.debugEnabled) {
+            def inputFiles = ""
+            inputs.files.each { f -> inputFiles += "$f " }
+            logger.debug "input files: $inputFiles"
+            def outputFiles = ""
+            outputs.files.each { f -> outputFiles += "$f " }
+            logger.debug "output files: $outputFiles"
+        }
+
+        args = [
+                '-war', gwtWarDir,
+                '-extra', gwtExtraDir,
+                '-logLevel', 'ERROR',   // The level of logging detail: ERROR, WARN, INFO, TRACE, DEBUG, SPAM or ALL (defaults to INFO)
+                '-localWorkers', 2,    // The number of local workers to use when compiling permutations
```

Comment:
After several benchmarks, I haven't been able to measure any improvement in the compilation time related to the number of `localWorkers`